gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -132,6 +139,23 @@ public TransportChannel getTransportChannel() throws IOException {
   }
 
   private TransportChannel createChannel() throws IOException {
+    ManagedChannel outerChannel;
+
+    if (poolSize == 1) {
+      outerChannel = createSubChannel();
+    } else {
+      ImmutableList.Builder<ManagedChannel> channels = ImmutableList.builder();
+
+      for (int i = 0; i < poolSize; i++) {
+        channels.add(createSubChannel());
+      }
+      outerChannel = new ChannelPool(channels.build());
+    }
+
+    return GrpcTransportChannel.create(outerChannel);
+  }
+
+  private ManagedChannel createSubChannel() throws IOException {
```

Comment:
I think "createSingleChannel()" would be better (since in the case where the pool size is 1, it isn't really a "sub" channel)

Reply:
done